src/borg/testsuite/archiver/create_cmd.py

```diff
@@ -650,7 +650,7 @@ def to_dict(borg_create_output):
         assert result["Modified files"] == 0
         # Archive a dir with two added files
         self.create_regular_file("testfile1", contents=b"test1")
-        time.sleep(1)  # testfile2 must have newer timestamps than testfile1
+        time.sleep(0.004)  # testfile2 must have newer timestamps than testfile1
```

Comment:
0.004 is the smallest that worked on my machine. Smaller numbers made the test pass randomly (sometimes it passes and sometimes it doesn't)

Reply:
did you find out why?

as i noted, modern fs should have timestamp with microseconds resolution or better.

Reply:
I checked the timestamps using the `stat` command in bash and the two files have the same timestamp down to the smallest decimal.
Maybe there is an OS level optimization that delays the creation of the files?

Reply:
@francoo98 can you check if an fsync and no delay solves it also?

@jdchristensen otoh, in case of the fsync taking >= 0.004s, we do not know why it worked (because of what it fundamentally does or because of how long that took).

Reply:
I did some tests, and using os.sync() instead of the time command works in practice.  @ThomasWaldmann, it's true that I can't be sure *why* it works, but it's still likely better than a hardcoded number that might fail on a slow system.

Reply:
(I also confirmed on my system that time.sleep(0.003) often produces files with identical time stamps.  This is suspicious, since ext4 should have better resolution than that.)

Reply:
Ah, there is an explanation here: https://stackoverflow.com/questions/70539589/file-seems-to-be-timestamped-before-it-is-created

It says that the time used for filesystem operations is only updated roughly every 10ms.  I just tried a loop that created 1000 files using `create_regular_file`, and there were only three different timestamps, almost exactly 4ms apart:
```
 Birth: 2022-10-16 17:15:15.092035097 -0400
 Birth: 2022-10-16 17:15:15.096035113 -0400
 Birth: 2022-10-16 17:15:15.100035129 -0400
```
(In some runs, the gap was 8ms, probably because the process had to give up the cpu for a bit.)

With a 0.003s sleep, there were ~800, and with a 0.004s sleep or `os.sync()` there were 1000.  I would propose either the sync method or a longer sleep (maybe 0.01s = 10ms?) to be safe.

Reply:
Interesting, would have expected it to be more fine grained. OK, let's use 0.01s for the sleep so we do not get spurious failures.

About the 4ms: maybe that timer that used to be 100Hz (10ms) is now 250Hz (4ms)?

https://stackoverflow.com/questions/14925568/lowering-linux-kernel-timer-frequency

But still, we want these tests to work in general, so we would prepare for worst case (100Hz).